### PR TITLE
[ADD] account_cost_center 8.0

### PR DESCRIPTION
**Pull request by astirpe**
Adding module Cost Center.
This module allows the user to link every invoice line to a cost center providing an extra dimension for the analysis.

### CODE REVIEW

**Review by nielsmiddeldorp — APPROVED** [review]

**Comment by andreschenkels** [comment]
:+1:  Works for me

File: account_cost_center/__openerp__.py
Comment (hbrunn) [inline account_cost_center/__openerp__.py]: don't use multiline strings when not necessary

File: account_cost_center/models/account_cost_center.py
Comment (hbrunn) [inline account_cost_center/models/account_cost_center.py]: `size` is obsolete for char fields, better remove them

File: account_cost_center/models/account_invoice.py
Comment (hbrunn) [inline account_cost_center/models/account_invoice.py]: why no new api here?

File: account_cost_center/models/account_invoice.py
Comment (hbrunn) [inline account_cost_center/models/account_invoice.py]: you can replace this hack with https://github.com/OCA/server-tools/tree/8.0/base_view_inheritance_extension by now

File: account_cost_center/models/account_invoice_line.py
Comment (hbrunn) [inline account_cost_center/models/account_invoice_line.py]: `None` already is the default for `dict.get`

File: account_cost_center/security/ir.model.access.csv
Comment (hbrunn) [inline account_cost_center/security/ir.model.access.csv:3]: is it really a good idea to give the less privileged users full permissions on cost centers?

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml]: let the orm generate a name if you don't pass anything informative

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml]: it suffices to attach the class to the label

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml:27]: empty elements on reserve?

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml]: see above

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml]: the `string` and `version` attributes don't do anything here imho

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [inline account_cost_center/views/account_cost_center.xml]: what's this default good for?

**Review by hbrunn — CHANGES_REQUESTED** [review]
some of the images violate [the branding rules](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules) of the OCA, in case you don't want to regenerate them, better remove them.

For all the `cost_center_id` fields in this module, I miss an index property. Is this a conscious choice?

File: account_cost_center/views/account_cost_center.xml
Comment (astirpe) [reply account_cost_center/views/account_cost_center.xml:27]: This was meant to be used by modules that extend this one.
I was creating some new modules that add pages on this form and I thought it was useful to have a notebook here.
Let say we remove this notebook, then if you have other two modules that add pages, you will need to couple those modules: one module adds the notebook, the other module extends from the first one adding pages.

I would prefer to keep this empty elements here in order to make inheritance easier.

File: account_cost_center/models/account_invoice.py
Comment (astirpe) [reply account_cost_center/models/account_invoice.py]: Let me try it ;)
I'll let you know!

File: account_cost_center/security/ir.model.access.csv
Comment (astirpe) [reply account_cost_center/security/ir.model.access.csv:3]: I keep the full permissions for 'group_account_user'. Do you agree?

File: account_cost_center/views/account_cost_center.xml
Comment (astirpe) [reply account_cost_center/views/account_cost_center.xml]: It seems this class doesn't have any effect here, I just removed the entire div.

File: account_cost_center/views/account_cost_center.xml
Comment (hbrunn) [reply account_cost_center/views/account_cost_center.xml:27]: for a module meant as base for a lot of others, that makes sense, yes

File: account_cost_center/security/ir.model.access.csv
Comment (hbrunn) [reply account_cost_center/security/ir.model.access.csv:3]: that's in line with having analytic accounts being fully accessible to account_user, so yes, makes sense

File: account_cost_center/models/account_move_line.py
Comment (hbrunn) [inline account_cost_center/models/account_move_line.py]: it's called `index` by now: https://github.com/OCA/OCB/blob/8.0/openerp/fields.py#L139

File: account_cost_center/models/account_invoice_line.py
Comment (hbrunn) [inline account_cost_center/models/account_invoice_line.py]: here too

**Comment by hbrunn** [comment]
nice! As soon as the indexes are fixed, we can merge in my opinion

**Review by hbrunn — APPROVED** [review]

**Comment by hbrunn** [comment]
nice! Any plans for further work or should I merge?

**Comment by astirpe** [comment]
For me this module is complete. Thanks!

**Comment by astirpe** [comment]
Thanks to you all for the reviews!
If you are interested, you can review the PR #442 containing the porting of account_cost_center to V9

**Pull request merged**
